### PR TITLE
using opentelemetry.io* namespace for extracing resource and attribute labels from k8s Api

### PR DESCRIPTION
**Pull request by Horiodino**
**Description:** 
i have added recognizeUserDefinedValues()  that will recognize the pods with pentelemetry.io/ prefix
#2181

**Testing:** i haven't tested it yet , im  not sure is it right so please tell any changes to made

### CODE REVIEW

File: pkg/instrumentation/sdk.go [hunk 1]
```diff
@@ -401,6 +402,54 @@ func chooseServiceVersion(pod corev1.Pod, index int) string {
...
+				ObjectMeta: metav1.ObjectMeta{
+					Name: constants.ReservedNamespace,
+				},
+			}
+			err = i.client.Create(ctx, &ns)
+			if err != nil {
+				return true, err
+			}
+			if err != nil {
+				return false, err
```
Comment (frzifus) [inline pkg/instrumentation/sdk.go:421]: It looks like this line is not reachable

File: pkg/instrumentation/sdk.go [hunk 2]
```diff
@@ -402,6 +402,54 @@ func chooseServiceVersion(pod corev1.Pod, index int) string {
 	return tag
 }
 
+// create/reserve namespace
+
+func (i *sdkInjector) createReservedNamespace(ctx context.Context) (bool, error) {
```
Comment (jaronoff97) [inline pkg/instrumentation/sdk.go:407]: why do we need to create a namespace here?

File: pkg/instrumentation/sdk.go [hunk 3]
```diff
@@ -402,6 +402,54 @@ func chooseServiceVersion(pod corev1.Pod, index int) string {
...
+			if err != nil {
+				return false, err
+			}
+		}
+	}
+	return true, nil
+}
+
+// recognizeUserDefinedValues recognizes user defined values assigned with opentelemetry.io/ prefix
+func (i *sdkInjector) recognizeUserDefinedValues(ctx context.Context, ns corev1.Namespace, pod corev1.Pod, deployment appsv1.Deployment, index int) map[string]string {
```
Comment (jaronoff97) [inline pkg/instrumentation/sdk.go:426]: could you add some tests in?

Comment (Horiodino) [reply pkg/instrumentation/sdk.go:426]: ok 👍

**Comment by Horiodino** [comment]
> @Horiodino per my last comment ([link](https://github.com/open-telemetry/opentelemetry-operator/issues/2181#issuecomment-1787506369)) there is no need to reach out to the kube API for any of this. Further, you're not using the functions you wrote and they are currently untested. If you'd like, I'd be happy to pair with you on this PR and we can work through it together.

sure :sparkles:

**Comment by jaronoff97** [comment]
feel free to reach out via the CNCF slack and we can set up some time

File: pkg/instrumentation/sdk.go [hunk 4]
```diff
@@ -30,11 +30,13 @@ import (
 	"go.opentelemetry.io/otel/attribute"
 	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
 	appsv1 "k8s.io/api/apps/v1"
+	v1obj "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (Snoopy666) [inline pkg/instrumentation/sdk.go]: .

**Review by Snoopy666 — APPROVED** [review]

**Comment by jaronoff97** [comment]
@Horiodino would you be able to fix the merge conflicts, fix the unit tests, and add a changelog? I would love to get this merged in, it's looking pretty good!

**Comment by jaronoff97** [comment]
@Horiodino bump on this, I'd love to get this in.

**Comment by Horiodino** [comment]
sorry for taking long , tried testing but that failed due to my system (previously using 32),
i will have fixed it now, i have added the changes now i will bw  creating a new branch and add the changes there other wise it will be mess as main has conflicts.

**Comment by Horiodino** [comment]
hey @jaronoff97  https://github.com/open-telemetry/opentelemetry-operator/pull/3010

**Pull request closed without merging**